### ISSUE TITLE
As a user I want to design an entity to be deployed as a smart contract table to specific blockchain so that I can design my dApp's domains

### ISSUE THREAD

**Issue by flowersinthesand** [lecle/aloxide#7]
To model dApps' domain, tools are required to design business' entities to be mapped to smart contract tables.

Given an oversimplified poll application, entities specification of poll and vote would be specified like this: 

`poll` table

- `id` number
- `hash` string
- `options` Array of string or JSON
- `creator` string

`vote` table

- `id` number
- `pollId` number
- `option` string
- `voter` string

In practice, with Aloxide these specifications are expected to be defined by the end-user and handled by the abstraction and bridge modules. 

With a configuration object style:

```js
const Poll = new Entity({
  name: "Poll",
  fields: [
    {name: "id", type: number},
    {name: "hash", type: string},
    {name: "options", type: Array},
    // Assume Signer type is given
    {name: "creator", type: Signer}
  ]
});
```

With a decorator pattern style which allows to use Plain Old JavaScript Object? (I personally prefer it over any other styles but think it can be added safely as an enhancement after the first GA release): 

```js
@entity
class Poll {
  @field
  id: number;
  @field
  hash: string;
  @field
  options: string[];
  @field
  creator: Signer;
}
```

Either way given `Poll` entity it's expected to be passed to the abstraction module configured with one of bridge module adapters.

```js
// Defined by the end-user
import Poll from './models/poll.js';
// Given by the abstraction module
import EntityManager from 'aloxide/abstraction';
// Given by the EOS bridge module
import EOSAdapter as adapter from 'aloxide/bridge/eos';

const adapter = new EOSAdapter({/* options */});
const entityManager = new EntityManager({adapter});

// Assumes that this method generates C++ code related to EOS smart contract table which is expected to be compiled and deployed by the end-user manually
await entityManager.generateCode(Poll, './eos/model/poll');

// Someday we could provide a method to deal with all procedures at once
// await entityManager.compileAndDeploy(Poll);
```

To sum up, to allow users to design their dApp domains with Aloxide, we need to 

- [ ] Figure out what is possible and what is not per blockchain: CAN, EOS and ICON.

Then, we should be able to decide what feature should be included in the abstraction so design interfaces and classes related to entity specification in the abstraction module.
